modules/data-entry/src/main/frontend/src/questionnaire/NumberQuestion.jsx

```diff
@@ -88,6 +116,37 @@ function NumberQuestion(props) {
   const [lowerLimit, setLowerLimit] = useState(initialValue[0]);
   const [upperLimit, setUpperLimit] = useState(initialValue[1]);
 
+  // The following is only used for non-range sliders.
+  // Default to an empty string, which results in a "no data"
+  // selection as close to 0 as possible within the valid range
+  const [sliderValue, setSliderValue] = useState(existingAnswer?.[1]?.value);
+
+  // The following is only used for ranged sliders.
+  // Setting a default of "" leads to an error, unlike the non-range case.
+  // Instead, default to a "no data" selection of both min and max being the lowest allowed value.
+  const sliderValues = [typeof(lowerLimit) === "undefined" ? minValue : Number(lowerLimit), typeof(upperLimit) === "undefined" ? minValue : Number(upperLimit)]
+
+  const isSlider = displayMode === "slider" && typeof minValue !== 'undefined' && typeof maxValue !== 'undefined';
+  const isRangeSelected = isRange && typeof(lowerLimit) != 'undefined' && !isNaN(+lowerLimit) && typeof(upperLimit) != 'undefined' && !isNaN(+upperLimit);
+  const isSingleSliderSelected = isSlider && typeof(sliderValue) != 'undefined' && !isNaN(+sliderValue);
+
+
+  // Marks at the minimum and maximum, as well as user specified intervals if provided
+  let sliderMarks = [{value: minValue, label: minValue}, {value: maxValue, label: maxValue}];
+  if (typeof(sliderMarkStep) !== "undefined") {
+    let i = minValue + sliderMarkStep;
+    while (i < maxValue) {
```

Comment:
vertical, minValue 0, maxValue 21, mark step 4 -> the labels for 20 and 21 are too close. I would not include the label for 20 in this case, just ...12, 16, 21.

```suggestion
    while (i <= maxValue - sliderMarkStep) {
```